src/dev/i18n/tasks/check_default_messages_in_files.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { glob } from 'glob';
+import { createFailError, isFailError, REPO_ROOT } from '@osd/dev-utils';
+import { ErrorReporter, filterConfigPaths, I18nConfig, normalizePath, readFileAsync } from '..';
+import { extractCodeMessages } from '../extractors';
+import { validateMessageNamespace } from '../extract_default_translations';
+
+function filterEntries(entries: string[], exclude: string[]) {
+  return entries.filter((entry: string) =>
+    exclude.every((excludedPath: string) => !normalizePath(entry).startsWith(excludedPath))
+  );
+}
+
+function addMessageToMap(
+  targetMap: Map<string, { message: string }>,
+  key: string,
+  value: { message: string },
+  reporter: ErrorReporter
+) {
+  const existingValue = targetMap.get(key);
+
+  if (targetMap.has(key) && existingValue!.message !== value.message) {
+    reporter.report(
+      createFailError(`There is more than one default message for the same id "${key}":
+"${existingValue!.message}" and "${value.message}"`)
+    );
+  } else {
+    targetMap.set(key, value);
+  }
+}
+
+async function extractMessagesFromFilesToMap(
+  files: string[],
+  targetMap: Map<string, { message: string }>,
+  config: I18nConfig,
+  reporter: any
```

Comment:
Is this `ErrorReporter | FailReport`?

Reply:
This is code I just copied from another part of i18n which uses `any` because different things can use it (I think). I don't think anyone other that `FailReport` would get here but lemme deal with that later. Thanks for this.